### PR TITLE
TE-10615 Skipped checks for deprecated methods and class

### PR DESCRIPTION
**Pull request by dmytro-dymarchuk**
- Developer(s): @dmytro-dymarchuk 

- Ticket: https://spryker.atlassian.net/browse/TE-10615

- Release Group: https://release.spryker.com/release-groups/view/3948

- merge: merge

**We will release this together with the bridge signature fixes.**

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ArchitectureSniffer   | patch (currently 0.x) {skip}  |                       |

-----------------------------------------

### Improvements

- Adjusted `LayerAccessRule` so it doesn't check deprecated classes.
- Adjusted `DirectoryNoModelRule` so it doesn't check deprecated classes.
- Adjusted `AllMethodIsPublicInFacadeRule` so it doesn't check deprecated methods.
- Adjusted `FacadeArgumentsNotAllowedUseEntityTransferRule` so it doesn't check deprecated methods.
- Adjusted `FacadeArgumentsRule` so it doesn't check deprecated methods.
- Adjusted `FacadeNoLogicRule` so it doesn't check deprecated methods.
- Adjusted `FacadeReturnValueRule` so it doesn't check deprecated methods.
- Adjusted `FacadeRule` so it doesn't check deprecated methods.
- Adjusted `FacadeSingleFactoryCallRule` so it doesn't check deprecated methods.
- Adjusted `PluginArgumentsNotAllowedUseEntityTransferRule` so it doesn't check deprecated classes.
- Adjusted `SpyEntityUsageRule` so it doesn't check deprecated methods.

### CODE REVIEW

**Review by dereuromark — APPROVED** [review]

File: src/Zed/Business/Facade/FacadeArgumentsRule.php [hunk 1]
```diff
@@ -48,6 +51,8 @@ public function apply(AbstractNode $node)
      */
     protected function applyRule(MethodNode $method)
     {
+        if ($this->isMethodDeprecated($method))
```
Comment (dereuromark) [inline src/Zed/Business/Facade/FacadeArgumentsRule.php:54]: This looks wrong, though.

**Comment by dmytro-dymarchuk** [comment]
Related to https://github.com/spryker/architecture-sniffer/issues/92

File: src/Zed/Business/Facade/FacadeArgumentsRule.php (hunk 1 above)
Comment (dmytro-dymarchuk) [reply src/Zed/Business/Facade/FacadeArgumentsRule.php]: Fixed.

**Comment by dmytro-dymarchuk** [comment]
Fixed cs and increased code coverage after approval.

**Comment by dereuromark** [comment]
We need to also fix the Bridge sniffer for
- mixed and other php8+ only types
- param signature detection (type in place if possible)

Can we pickyback that here or should we open a 2nd PR and make this merge only for now?

**Pull request merged**